homeassistant/components/shopping_list/__init__.py

```diff
@@ -144,7 +144,7 @@ def async_add(self, name):
         """Add a shopping list item."""
         item = {"name": name, "id": uuid.uuid4().hex, "complete": False}
         self.items.append(item)
-        self.hass.async_add_job(self.save)
+        self.hass.async_add_executor_job(self.save)
```

Comment:
I'm wondering if there's a reason why we're not in a coroutine and not awaiting this job? Seems like it would be safer to await it.

Reply:
I'm going to mark this as a draft while I do some testing.

Reply:
Adjusted everything to be awaited and ran though it.  Verified I could still add remove and update, no errors in the log, and no errors on the console.